apps/webservice/src/app/[workspaceSlug]/_components/EditDeploymentDialog.tsx

```diff
@@ -0,0 +1,145 @@
+"use client";
+
+import React, { useState } from "react";
+import { useRouter } from "next/navigation";
+import { useForm } from "react-hook-form";
+import { z } from "zod";
+
+import { Button } from "@ctrlplane/ui/button";
+import { CopyButton } from "@ctrlplane/ui/copy-button";
+import {
+  Dialog,
+  DialogContent,
+  DialogDescription,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+  DialogTrigger,
+} from "@ctrlplane/ui/dialog";
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormLabel,
+} from "@ctrlplane/ui/form";
+import { Input } from "@ctrlplane/ui/input";
+import { Textarea } from "@ctrlplane/ui/textarea";
+
+import { api } from "~/trpc/react";
+
+const deploymentForm = z.object({
+  id: z.string().uuid(),
+  name: z.string().min(3).max(255),
+  slug: z.string().min(3).max(255),
+  description: z.string().optional(),
+});
+
+type DeploymentFormValues = z.infer<typeof deploymentForm>;
+
+export const EditDeploymentDialog: React.FC<
+  DeploymentFormValues & { children?: React.ReactNode }
+> = ({ id, name, slug, description, children }) => {
+  const router = useRouter();
+  const update = api.deployment.update.useMutation();
+  const [open, setOpen] = useState(false);
+
+  const form = useForm<DeploymentFormValues>({
+    defaultValues: { id, name, slug, description },
+  });
```

Comment:
```suggestion
  const form = useForm({
    schema: deploymentForm,
    defaultValues: { id, name, slug, description },
  });
```

Reply:
Needed

```ts
  const form = useForm({
    resolver: zodResolver(deploymentForm),
    defaultValues: { id, name, slug, description },
  });
```

Reply:
You do not, you are importing the wrong useForm

Reply:
import from ctrlplane ui